### PR TITLE
[1.12] Preserve TypeURL in gRPC->gRPC service invocation 

### LINKED ISSUE #7012: Error calling gRPC Service Invocation with .NET
## In what area(s)?
/area runtime
> /area operator
> /area placement
> /area docs
> /area test-and-release

## What version of Dapr?

1.12.0-rc.x

## Expected Behavior

gRPC invocation should work as it did in 1.11.

## Actual Behavior

The .NET gRPC invocation example (non-proxy) fails to complete a request. The issue appears to be happening with the `Unpack` of an `Any` field in the proto's message. 

```
== APP == info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
== APP ==       Request starting HTTP/2 POST http://127.0.0.1:5050/dapr.proto.runtime.v1.AppCallback/OnInvoke application/grpc -
== APP == info: Microsoft.AspNetCore.Routing.EndpointMiddleware[0]
== APP ==       Executing endpoint 'gRPC - /dapr.proto.runtime.v1.AppCallback/OnInvoke'
== APP == fail: Grpc.AspNetCore.Server.ServerCallHandler[6]
== APP ==       Error when executing service method 'OnInvoke'.
== APP ==       Google.Protobuf.InvalidProtocolBufferException: Full type name for Transaction is Transaction; Any message's type url is
== APP ==          at Google.Protobuf.WellKnownTypes.Any.Unpack[T]()
== APP ==          at GrpcServiceSample.BankingService.OnInvoke(InvokeRequest request, ServerCallContext context) in C:\Users\halspang\Workspace\dotnet-sdk\examples\AspNetCore\GrpcServiceSample\Services\BankingService.cs:line 72
== APP ==          at Grpc.Shared.Server.UnaryServerMethodInvoker`3.AwaitInvoker(Task`1 invokerTask, GrpcActivatorHandle`1 serviceHandle)
== APP ==          at Grpc.Shared.Server.UnaryServerMethodInvoker`3.AwaitInvoker(Task`1 invokerTask, GrpcActivatorHandle`1 serviceHandle)
== APP ==          at Grpc.AspNetCore.Server.Internal.CallHandlers.UnaryServerCallHandler`3.HandleCallAsyncCore(HttpContext httpContext, HttpContextServerCallContext serverCallContext)
== APP ==          at Grpc.AspNetCore.Server.Internal.CallHandlers.ServerCallHandlerBase`3.<HandleCallAsync>g__AwaitHandleCall|8_0(HttpContextServerCallContext serverCallContext, Method`2 method, Task handleCall)
== APP == info: Microsoft.AspNetCore.Routing.EndpointMiddleware[1]
== APP ==       Executed endpoint 'gRPC - /dapr.proto.runtime.v1.AppCallback/OnInvoke'
== APP == info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
== APP ==       Request finished HTTP/2 POST http://127.0.0.1:5050/dapr.proto.runtime.v1.AppCallback/OnInvoke application/grpc - - 200 0 application/grpc 119.7066ms
```

## Steps to Reproduce the Problem

Run the .NET [gRPC service invocation example](https://github.com/dapr/dotnet-sdk/blob/master/examples/Client/ServiceInvocation/README.md)

## Release Note

RELEASE NOTE: **FIX** Invocation error in gRPC service to service call

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Fixes #7012

Also fixes: in gRPC->gRPC/HTTP service invocation, data in the first chunk was included twice

### CODE REVIEW

**Comment by artursouza** [comment]
Watching test results to merge it.

File: pkg/messaging/direct_messaging.go [hunk 1]
```diff
@@ -336,6 +336,21 @@ func (d *directMessaging) invokeRemoteStream(ctx context.Context, clientV1 inter
 	}()
 	r := req.RawData()
 	reqProto := req.Proto()
+
+	// If there's a message in the proto, we remove it from the message we send to avoid sending it twice
+	// However we need to keep a reference to those bytes, and if needed add them back, to retry the request with resiliency
+	// We re-add it when the method ends to ensure we can perform retries
+	messageData := reqProto.GetMessage().GetData()
```
Comment (yaron2) [inline pkg/messaging/direct_messaging.go:343]: Can `GetMessage()` or `GetData()` return nil?

Comment (ItalyPaleAle) [reply pkg/messaging/direct_messaging.go:343]: All the `Get*` methods have built-in nil checks:

https://github.com/dapr/dapr/blob/fa55f33e4cf7b95121a361adbbacf20ad0d9b58c/pkg/proto/common/v1/common.pb.go#L330-L342

**Review by yaron2 — APPROVED** [review]

**Pull request merged**